src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -493,56 +530,134 @@ private Float[][] PostProcess(Float[][] y, Float[] sigma, Float[] z, int d, int
             return y;
         }
 
-        private ColumnType[] InitColumnTypes()
+        protected override IRowMapper MakeRowMapper(ISchema schema) => new Mapper(this, Schema.Create(schema));
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
-            Host.Assert(Infos.Length == _transformInfos.Length);
-            var types = new ColumnType[Infos.Length];
-            for (int i = 0; i < _transformInfos.Length; i++)
-                types[i] = new VectorType(NumberType.Float, _transformInfos[i].Rank);
-            Metadata.Seal();
-            return types;
+            ValidatePcaInput(Host, inputSchema.GetColumnName(srcCol), inputSchema.GetColumnType(srcCol));
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        internal static void ValidatePcaInput(IHost host, string name, ColumnType type)
```

Comment:
>static [](start = 17, length = 6)

any reason why it's static?

Reply:
it reflects the fact that the PCA check doesn't depends any instance state. It only cares about a column type. It also gets used inside the Mapper. 

---
In reply to: [227576143](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227576143) [](ancestors = 227576143)